src/main/java/net/dv8tion/jda/internal/requests/restaction/GuildScheduledEventActionImpl.java

```diff
@@ -0,0 +1,201 @@
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.entities.GuildImpl;
+import net.dv8tion.jda.internal.requests.Route;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.Helpers;
+import okhttp3.RequestBody;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+public class GuildScheduledEventActionImpl extends AuditableRestActionImpl<GuildScheduledEvent> implements GuildScheduledEventAction
+{
+    protected final Guild guild;
+    protected String name, description;
+    protected Icon image;
+    protected long channelId;
+    protected String location;
+    protected OffsetDateTime startTime, endTime;
+    protected int entityType;
+
+    public GuildScheduledEventActionImpl(Guild guild)
+    {
+        super(guild.getJDA(), Route.Guilds.CREATE_SCHEDULED_EVENT.compile(guild.getId()));
+        this.guild = guild;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventActionImpl setCheck(BooleanSupplier checks)
+    {
+        return (GuildScheduledEventActionImpl) super.setCheck(checks);
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventActionImpl timeout(long timeout, @Nonnull TimeUnit unit)
+    {
+        return (GuildScheduledEventActionImpl) super.timeout(timeout, unit);
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventActionImpl deadline(long timestamp)
+    {
+        return (GuildScheduledEventActionImpl) super.deadline(timestamp);
+    }
+
+    @Nonnull
+    @Override
+    public Guild getGuild()
+    {
+        return guild;
+    }
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventActionImpl setName(@Nonnull String name)
+    {
+        Checks.notBlank(name, "Name");
+        name = name.trim();
+        Checks.notEmpty(name, "Name");
+        Checks.notLonger(name, GuildScheduledEvent.MAX_NAME_LENGTH, "Name");
+        this.name = name;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventActionImpl setDescription(@Nullable String description)
+    {
+        if (description != null)
+            Checks.notLonger(description, GuildScheduledEvent.MAX_DESCRIPTION_LENGTH, "Description");
+        this.description = description;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setImage(@Nullable Icon icon)
+    {
+        this.image = icon;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setLocation(@Nonnull StageChannel stageChannel)
+    {
+        this.channelId = stageChannel.getIdLong();
```

Comment:
Same null checks as in the manager missing on these setters

Reply:
I removed these setters because the method takes them as parameters now. Null checks should be there now